lib/authentication/auth_web.js

```diff
@@ -74,29 +77,47 @@ function AuthWeb(connectionConfig, httpClient, webbrowser) {
     // Use a free random port and set to no backlog
     server.listen(0, 0);
 
-    // Step 1: query Snowflake to obtain SSO url
-    const ssoData = await ssoUrlProvider.getSSOURL(authenticator,
-      serviceName,
-      account,
-      server.address().port,
-      username,
-      host);
-
-    proofKey = ssoData['proofKey'];
-
+    if (connectionConfig.getDisableConsoleLogin()) {
+      // Step 1: query Snowflake to obtain SSO url
+      const ssoData = await ssoUrlProvider.getSSOURL(authenticator,
```

Comment:
is this `await` needed? This is not an async function right?

Reply:
Yes we need this becasuse the return data from getSSOURL is a Promise. If we did not use await, ssoData will be a Promise<pending>, not a data.